src/rgw/rgw_rest_s3.cc

```diff
@@ -2116,6 +2117,9 @@ int RGWCopyObj_ObjStore_S3::get_params()
     /* can only copy object into itself if replacing attrs */
     ldout(s->cct, 0) << "can't copy object into itself if not replacing attrs"
 		     << dendl;
+    s->err.message = "This copy request is illegal because it is trying to copy "
+                     "an object to itself without changing the object's metadata, "
+                     "storage class, website redirect location or encryption attributes.";
```

Comment:
ditto. Otherwise, do you have a justification for a different message ? I see that the original message is the brief of your expanded/detailed version.

Reply:
@joscollin Thanks for your suggestion. I just keep the response error message same with AWS S3 for compatibility.

Reply:
@joscollin Do you still think I should keep s->err.message same with ldout? If so, I can change my commit in your way~ Thank you~

Reply:
I would say, you can keep the `s->err.message` text the same to be compatible with AWS S3. But I was suggesting that you could [change it this way](https://github.com/ZVampirEM77/ceph/blob/f701ad80e793cd442b10d3a6f198f41d80389ebd/src/rgw/rgw_rest_s3.cc#L2802), instead of two different texts. What do you think ?

Reply:
OK